SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -75,26 +103,24 @@ void USpatialLoadBalanceEnforcer::ProcessQueuedAclAssignmentRequests()
 
 	for (WriteAuthAssignmentRequest& Request : AclWriteAuthAssignmentRequests)
 	{
-		static const int32 WarnOnAttemptNum = 5;
-		if (Request.ProcessAttempts >= WarnOnAttemptNum)
-		{
-			UE_LOG(LogSpatialLoadBalanceEnforcer, Warning, TEXT("Failed to process WriteAuthAssignmentRequest with EntityID: %lld. Process attempts made: %d"), Request.EntityId, Request.ProcessAttempts);
-		}
-
-		Request.ProcessAttempts++;
-
-		// TODO - if some entities won't have the component we should detect that before queueing the request.
-		// Need to be certain it is invalid to get here before receiving the AuthIntentComponent for an entity, then we can check() on it.
-		const AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(Request.EntityId);
+ 		const AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(Request.EntityId);
 		if (AuthorityIntentComponent == nullptr)
 		{
-			UE_LOG(LogSpatialLoadBalanceEnforcer, Warning, TEXT("Detected entity without AuthIntent component. EntityId: %lld"), Request.EntityId);
-			continue;
+			// TODO(zoning): This is possible if the ACL update lags behind the intent update. Evaluate whether this is a serious problem.
+			UE_LOG(LogSpatialLoadBalanceEnforcer, Warning, TEXT("(%s) Entity without AuthIntent component will not be processed. EntityId: %lld"), *WorkerId, Request.EntityId);
+			CompletedRequests.Add(Request.EntityId);
+			return;
 		}
 
 		const FString* OwningWorkerId = VirtualWorkerTranslator->GetPhysicalWorkerForVirtualWorker(AuthorityIntentComponent->VirtualWorkerId);
 		if (OwningWorkerId == nullptr)
 		{
+			const int32 WarnOnAttemptNum = 5;
```

Comment:
When can the above conditional fail? Due to the startup ops filtering, we should have received the full map of virtual worker -> physical worker IDs before processing any entity component updates / authority changes (how this function is triggered) so there shouldn't be a case where the physical worker isn't set?

Reply:
Right now the IsReady check is whether any mapping has been received, not necessarily that all mappings for all workers have been received. Also, this could happen (once we support it) when a worker has crashed and the replacement hasn't been selected.

Reply:
I think in that case it would set it to the old virtual worker ID. Then it would be the responsibility of the enforcers to change all ACLs with the corresponding intent when they see an update to the map.

Reply:
Not necessarily all mappings being received and worker replacing definitely makes sense as reasons for this check - could we add a comment as I think it's really useful content? 

In the long term, I agree with Sami. When we have a worker crashing and have the translator learn of the new physical worker ID of the restarted worker, it'll map the according virtual worker ID. I imagine we'll have it inform the enforcer of the new update so the enforcer can update all relevant entity ACLs. In this circumstance (which is similar to the initial mapping for that virtual worker not yet having been received), we'll need a way for the enforcer to find all entities with the virtual worker ID so it can apply the update. Since we'll need to have that, we'll also have a more efficient solution to entities who are ticking in `AclWriteAuthAssignmentRequests` but can't find their translator mapping entry. So then, in future, if this check fails, we can just delete the entry in `AclWriteAuthAssignmentRequests` because we know the enforcer will update it when it learns the new mapping for that virtual worker? Does that make sense? I think it makes sense? 😅

Reply:
I think that makes sense, but the entire argument is built on hypothetical code that hasn't been written. For now, I'm going to put a TODO in to revisit when we do get around to adding restarting of virtual workers.